File: Data Structures & Algorithms/sort-colors/submission-0.java
class Solution {
    public void sortColors(int[] nums) {
        int[] count = new int[3];

        for (int n : nums) {
            count[n]++;
        }

        for (int i = 0; i < nums.length; i++) {
            if (count[0] != 0) {
                nums[i] = 0;
                count[0]--;
            } else if (count[1] != 0) {
                nums[i] = 1;
                count[1]--;
            } else {
                nums[i] = 2;
                count[2]--;
            }
        }
    }
}

File: Data Structures & Algorithms/combination-sum-iv/submission-0.py
class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        memo = defaultdict(int)

        def dfs(remainder):
            if remainder == 0:
                return 1
            if remainder < 0:
                return 0
            if memo[remainder]:
                return memo[remainder]
            
            total = 0
            for n in nums:
                total += dfs(remainder - n)
            
            memo[remainder] = total
            return total
        
        return dfs(target)